src/goalServices/changeGoalStatus.ts

```diff
@@ -10,6 +10,39 @@ interface GoalStatusChangeParams {
   transaction?: Sequelize.Transaction;
 }
 
+export async function changeGoalStatusWithSystemUser({
```

Comment:
do you want to DRY this up and combine the bottom half of the two functions? 

I.E. both call basically this exact thing, maybe they could call an "updateGoalStatus" function or something:

```js
 // Change goal status.
  await db.GoalStatusChange.create({
    goalId: goal.id,
    userId: null, // For now we will use null to prevent FK constraint violation.
    userName: 'system',
    userRoles: null,
    oldStatus: goal.status,
    newStatus,
    reason,
    context,
  });

  // Reload goal.
  await goal.reload();

  //  Return goal.
  return goal;
```

I'll approve

Reply:
I thought about this but just decided to keep the separate.